Sources/Basics/Observability.swift

```diff
@@ -0,0 +1,292 @@
+        } else if let convertible = error as? DiagnosticDataConvertible {
+            message = "\(convertible.diagnosticData)"
+        } else {
+            message = "\(error)"
+        }
+        return Self(severity: .error, message: message, data: data)
+    }
+
+    public static func warning(_ message: String, data: CustomStringConvertible? = .none) -> Self {
+        Self(severity: .warning, message: message, data: data)
+    }
+
+    public static func warning(_ message: CustomStringConvertible, data: CustomStringConvertible? = .none) -> Self {
+        Self(severity: .warning, message: message.description, data: data)
+    }
+
+    public static func info(_ message: String, data: CustomStringConvertible? = .none) -> Self {
+        Self(severity: .info, message: message, data: data)
+    }
+
+    public static func info(_ message: CustomStringConvertible, data: CustomStringConvertible? = .none) -> Self {
+        Self(severity: .info, message: message.description, data: data)
+    }
+
+    public static func note(_ message: CustomStringConvertible, data: CustomStringConvertible? = .none) -> Self {
+        // 👀 do we need info and note?
+        Self(severity: .note, message: message.description, data: data)
+    }
+
+    public enum Severity: Equatable {
+        case error
+        case warning
+        // 👀 do we need info and note?
+        case info
+        case note
+    }
+
+    public static func == (lhs: DiagnosticMessage, rhs: DiagnosticMessage) -> Bool {
+        if lhs.severity != rhs.severity {
+            return false
+        }
+        if lhs.message != rhs.message {
+            return false
+        }
+        if lhs.data?.description != rhs.data?.description {
+            return false
+        }
+        return true
+    }
+}
+
+// 👀 TODO: this could/should be changed to more structured metadata model
+public protocol DiagnosticsContext: CustomStringConvertible {}
+
+public struct StringDiagnosticsContext: DiagnosticsContext{
+    public private (set) var description: String
+
+    public init(_ description: String) {
+        self.description = description
+    }
+}
+
+@available(*, deprecated, message: "temporary for transition DiagnosticsEngine -> DiagnosticsEmitter")
+extension Diagnostic {
+    init?(_ diagnostic: TSCDiagnostic) {
+        switch diagnostic.behavior {
+        case .error:
+            self = .init(context: DiagnosticLocationWrapper(diagnostic.location), message: .error(diagnostic.message.text, data: diagnostic.message.data))
+        case .warning:
+            self = .init(context: DiagnosticLocationWrapper(diagnostic.location), message: .warning(diagnostic.message.text, data: diagnostic.message.data))
+        case .note:
+            self = .init(context: DiagnosticLocationWrapper(diagnostic.location), message: .note(diagnostic.message.text, data: diagnostic.message.data))
+        case .remark:
+            // 👀 remark mapped to info here, do we need all these levels?
```

Comment:
I would say we could map this to a note.  I never understood the need to distinguish notes from remarks.  It's basically just "non-warning info".